File: go/leetcode/medium/274-H 指数.go
package main

import "sort"

func hIndex(citations []int) int {
	sort.Ints(citations)
	temp := 0
	for i, citation := range citations {
		if citation >= len(citations)-i {
			temp = len(citations) - i
			break
		}
	}
	return temp
}

func main() {

}

File: python/leetcode/algorithms/medium/274-H 指数.py
from typing import List


class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        temp = 0
        for i, citation in enumerate(citations):
            if citation == len(citations) - i:
                temp = len(citations) - i
                break
            if citation > len(citations) - i:
                temp = len(citations) - i
                break
        return temp


citations = [2, 3, 4, 1, 7, 6]
s = Solution()
print(s.hIndex(citations))
